packages/ngtools/webpack/src/ivy/plugin.ts

```diff
@@ -422,13 +421,34 @@ export class AngularWebpackPlugin {
     }
 
     // Update semantic diagnostics cache
+    const affectedFiles = new Set<ts.SourceFile>();
     while (true) {
-      const result = builder.getSemanticDiagnosticsOfNextAffectedFile(undefined, (sourceFile) =>
-        ignoreForDiagnostics.has(sourceFile),
-      );
+      const result = builder.getSemanticDiagnosticsOfNextAffectedFile(undefined, (sourceFile) => {
+        // If the affected file is a TTC shim, add the shim's original source file.
+        // This ensures that changes that affect TTC are typechecked even when the changes
+        // are otherwise unrelated from a TS perspective and do not result in Ivy codegen changes.
+        // For example, changing @Input property types of a directive used in another component's
+        // template.
+        if (ignoreForDiagnostics.has(sourceFile)) {
+          // This file name conversion relies on internal compiler logic and should be converted
+          // to an official method when available.
+          const originalFilename = sourceFile.fileName.replace(/\.ngtypecheck(\.tsx?)$/, '$1');
```

Comment:
The shim will always be `.ts`, even if the original was `.tsx`. So the capture group won't help here (and the reverse mapping is therefore ambiguous, although this can be resolved based on `originalSourceFile` ends up being `undefined`)

Reply:
And possibly we should only do this for files that actually end in `.ngtypecheck.ts`? As otherwise we'd add the shim source file to the list of `affectedFiles`.